app/src/test/java/com/codesoom/assignment/application/UserCommandServiceTest.java

```diff
@@ -0,0 +1,258 @@
+            void it_throws_exception() {
+                assertThatThrownBy(
+                        () -> userCommandService.deleteUser(INVALID_USER_ID)
+                ).isExactlyInstanceOf(UserNotFoundException.class);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("deleteUsers 메서드는")
+    @SpringBootTest
+    class Describe_deleteUsers {
+
+        @Autowired
+        private UserRepository userRepository;
+
+        @Nested
+        @DisplayName("저장되어있는 user 의 id가 주어지면 ")
+        class Context_with_existing_user_id {
+            private Set<Long> deletedIds;
+
+            @BeforeEach
+            void setUp() {
+                User savedUser1 = userCommandService.createUser(
+                        UserCreateRequest.builder()
+                                .email("a@a.com")
+                                .name("김 코")
+                                .password("a")
+                                .build()
+                );
+                User savedUser2 = userCommandService.createUser(
+                        UserCreateRequest.builder()
+                                .email("b@b.com")
+                                .name("김 코")
+                                .password("b")
+                                .build()
+                );
+
+                deletedIds = Set.of(savedUser1.getId(), savedUser2.getId());
+            }
+
+            @Test
+            @DisplayName("user 들을 삭제하고 삭제 보고서를 리턴한다")
+            void it_returns_delete_user_report() {
+
+                UserDeleteReport userDeleteReport = userCommandService.deleteUsers(deletedIds);
+
+                assertThat(userRepository.findAllById(deletedIds)).isEmpty();
+                assertThat(userRepository.findAllById(userDeleteReport.getDeletedSuccessIds())).isEmpty();
+
+                assertThat(userDeleteReport.getDeletedSuccessIds()).isEqualTo(deletedIds);
+                assertThat(userDeleteReport.getDeletedFailIds()).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("저장되어있지 않은 user 의 id가 주어지면 ")
+        class Context_with_non_existence_user_id {
+            private Long deleteId;
+
+            @BeforeEach
+            void setUp() {
+                User savedUser = userCommandService.createUser(
+                        UserCreateRequest.builder()
+                                .email("a@a.com")
+                                .name("김 코")
+                                .password("a")
+                                .build()
+                );
+                deleteId = savedUser.getId();
+            }
+
+            @Test
+            @DisplayName("user 들을 삭제하고 삭제 보고서를 리턴한다")
+            void it_returns_delete_user_report() {
+                UserDeleteReport userDeleteReport = userCommandService.deleteUsers(Set.of(deleteId, INVALID_USER_ID));
+
+                assertThat(userRepository.findById(deleteId))
+                        .as("삭제되어야할" + deleteId + "가 삭제됐는지 확인")
```

Comment:
여기에서는 메시지를 단호하게 적어보세요.

```suggestion
                        .as("삭제한 사용자를 조회했을 때 결과가 없어야 합니다.")
```